docs/architecture/adr-021-restricted-block-size.md

```diff
@@ -0,0 +1,156 @@
+[#1743](https://github.com/celestiaorg/celestia-app/pull/1743).
+
+### Introduce a new Governance Parameter
+
+The second suggestion is to create a new governance parameter,
+`GovMaxSquareSize`, which governance could set. Now the protocol would use the
+lowest of 4 values to determine the size of a block at a given height. This has
+the benefit of allowing for the most flexibility in terms of options, however it
+has the downside of exposing a rather technical parameter to governance. Full
+implementation in [#1772](https://github.com/celestiaorg/celestia-app/pull/1772)
+
+### Use `MaxBytes` Governance Parameter to Limit Square Size
+
+The third suggested solution is to use the existing governance parameter to also
+contribute to limiting the square size. This would allow for governance to set
+the `MaxBytes` value, and then the application would use that value along with
+the `MaxSquareSize` to determine the `GovMaxSquareSize` which would be used to
+create the square. If governance wanted to limit the square size, then it would
+set a value that is smaller than or equal to the optimal value of a given square
+size. This has the benefit of not exposing another parameter to governance, but
+it makes the result of the currently exposed parameter, `MaxBytes`, more
+complicated. It also eliminates the possibility to allow for small blocks with a
+lot of encoding overhead, since the only way to increase the square size is to
+also increase the `MaxBytes`.Full implementation in
+[#1765](https://github.com/celestiaorg/celestia-app/pull/1765)
+
+Note that there is technically a fourth solution, where the `GovMaxSquareSize`
+is added, and it determines the value for `MaxBytes`. The result of this
+solution is effectively identical to just adding a new `GovMaxSquareSize`
+parameter.
+
+## Decision
+
+Option 2: Introduce a new parameter, `GovMaxSquareSize`. After implemented, the above chart will look like:
+
+| Parameter        | Size   | Unit   | Description  | Value Control |
+|------------------|--------|--------|--------------|---------------|
+| `MaxBlockSizeBytes` | 100 | MiB | Maximum total size of the protobuf encoded block, a hard coded constant acting as a cap for `MaxBytes`. | Hard coded |
+| `MaxBytes`     |  ~1.8 | MiB | Determines the valid size of the entire protobuf encoded block. Is a governance modifiable parameter and is capped by `MaxBytes`. Used to regulate the amount of data gossiped in the consensus portion of the network (using the current block gossiping mechanism). | Modifiable |
+| `MaxSquareSize` | ~7.5 | MiB | Determines the maximum size of the original data square. Used to regulate storage requirements for celestia-node, and the size of the data availability header. Default set to 128. | Hard coded |
+| `GovMaxSquareSize`     |  ~1.8 | MiB | Governance modifiable parameter that determines valid square sizes. Must be smaller than the `MaxSquareSize`. Default set to 64. | Modifiable |
```

Comment:
```suggestion
| `GovMaxSquareSize`     |  ~1.8 | MiB | Governance modifiable parameter that determines the max valid square size. Must be smaller than the `MaxSquareSize`. Default set to 64. | Modifiable |
```

Reply:
merging for the sake of getting this through, but this does make sense to change